pandas/core/strings.py

```diff
@@ -472,9 +472,17 @@ def str_extract(arr, pat, flags=0):
     from pandas.core.index import Index
 
     regex = re.compile(pat, flags=flags)
-    # just to be safe, check this
+
     if regex.groups == 0:
         raise ValueError("This pattern contains no groups to capture.")
+    
+    if isinstance(arr, Index):
+        if 1 < regex.groups:
+            raise ValueError("only one regex group is supported with Index")
```

Comment:
In my opinion we should not return Index, even when it is called on an Index.

extract/extractall should always return DataFrame, and then if the user wants he can create an Index or MultiIndex.

Again this should be easier to maintain and easier for users to understand.